src/app/shared/role-guard-service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { CouchService } from './couchdb.service';
+import { UserService } from './user.service';
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+
+@Injectable()
+export class RoleService {
+
+  constructor(private couchService: CouchService, private userService: UserService, private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): any {
+    if (route.url.toString() === 'add') {
+      if (this.userService.get().roles.length !== 0 || this.userService.get().isUserAdmin) {
```

Comment:
As I wrote in my previous feedback, this approach will cause us a lot of headaches if we change security settings on different databases.  We should base route access off the security that are currently on CouchDB, not hardcoded in the app like this.